include/solvers/navier_stokes_scratch_data.h

```diff
@@ -839,14 +914,175 @@ class NavierStokesScratchData : public PhysicsScratchDataBase
+  void
+  reinit_particle_fluid_interactions(
+    const typename DoFHandler<dim>::active_cell_iterator & /*cell*/,
+    const VectorType & /*current_solution*/,
+    const VectorType                      &previous_solution,
+    const VectorType                      &void_fraction_solution,
+    const Particles::ParticleHandler<dim> &particle_handler,
+    const DoFHandler<dim>                 &dof_handler,
+    const DoFHandler<dim>                 &void_fraction_dof_handler)
+  {
+    pic = particle_handler.particles_in_cell(this->fe_values.get_cell());
+
+    // Calculate the particles properties
+    double total_particle_volume = 0;
+    total_particle_volume        = extract_particle_properties();
+
+    calculate_cell_void_fraction(total_particle_volume);
+
+    if (number_of_particles == 0)
+      return;
+
+    // Calculate the average particle velocity within the cell
+    average_particle_velocity = average_particle_velocity / number_of_particles;
+
+    Quadrature<dim> q_particles =
+      calculate_particle_location_quadrature_points();
+    calculate_fluid_fields_at_particle_location(dof_handler,
+                                                q_particles,
+                                                previous_solution);
+
+    // Create a quadrature for the void fraction that is based on the particle
+    // reference location
+    if (this->interpolated_void_fraction)
+      {
+        calculate_void_fraction_at_particle_location(q_particles,
+                                                     void_fraction_dof_handler,
+                                                     void_fraction_solution);
+      }
+    calculate_fluid_properties_at_particle_location();
+    calculate_force_parameters_at_particle_location();
+  }
+
+  // Version of the function when VOF is used with CFD-DEM
+  /** @brief Calculates the variables needed to compute the particle fluid interactions
+   * in the vans equations.This version of the function is used when VOF is used
+   * with CFD-DEM.
+   *
+   * @param previous_solution The solution at the previous time step for the fluid's velocity
+   * and pressure
+   *
+   * @param void_fraction_solution The void fraction value calculated with one of the methods of
+   * the VoidFractionBase class.
+   *
+   * @param particle_handler The particle handler object that stores and manages the
+   *  particles in the simulations
+   *
+   * @param dof_handler The DoFHandler object for the Navier-Stokes problem
+   *
+   * @param void_fraction_dof_handler The DoFHandler object for the void fraction calculation
+   *
+   * @param dof_handler_vof The DoFHandler object for the VOF equation
+   *
+   * @param current_filtered_solution The present value of the phase fraction at the dofs of the
+   * cell
+   */
+  template <typename VectorType>
+  void
+  reinit_particle_fluid_interactions(
+    const typename DoFHandler<dim>::active_cell_iterator & /*cell*/,
+    const VectorType & /*current_solution*/,
```

Comment:
If cell is never used, I would consider removing it as an argument. to this function then.